homework-2/script/maim.js

```diff
@@ -0,0 +1,318 @@
+    const updateUserBtn = document.querySelector('.btn-dark');
+    const showPostForm = document.querySelector('.form-btn');
+    const modalCreateUser = document.querySelector('.modal-create-user');
+    const postBtn = document.querySelector('.post-btn');
+    const newNameData = document.querySelector('.create-name');
+    const newCompanyData = document.querySelector('.create-company');
+    const newAddressData = document.querySelector('.create-address');
+    const newCityData = document.querySelector('.create-city');
+    const newCountyData = document.querySelector('.create-country');
+    const allCheckbox = document.querySelector('.box-checkbox');
+    const paginationPage = document.querySelector('.pagination')
+    let secondaryCheckbox;
+    let deleteBtn;
+    let checkboxBtn;
+
+    const userFilterCompany = document.querySelector('.user-company');
+    const userFilterName = document.querySelector('.user-contact');
+    const filterUserBtn = document.querySelector('.filter-user');
+    const userFilterAddress = document.querySelector('.user-address');
+
+    let modalNumber;
+    let dataCompany = [];
+    let dataName = [];
+    let dataAddress = [];
+    let dataIdUser = [];
+    let navigationBtn = [];
+
+
+    let url = 'http://localhost:3000/users';
+
+    const getAllUsers = async ( url ) => {
+        const result = await fetch( url );
+
+        if( !result.ok ) {
+            throw new Error(`Could not fetch ${url}, status: ${result.status}`)
+        }
+
+        return await result.json();
+    }
+
+
+
+    const createUser = async ( obj ) => {
+        const create = await fetch( url, {
+            method: 'POST',
+            body: JSON.stringify(obj),
+              headers: {
+                'Content-type': 'application/json; charset=UTF-8',
+              },
+        });
+        
+        if (!create.ok) {
+            throw new Error(`Someting went wrong, Status code ${create.status}`)
+        }
+        return await create.json();
+    }
+
+
+    
+    const deleteUser = async (id) => {
+        const response = await fetch(`${url}/${id}`, {
+            method: 'DELETE'
+        });
+
+        if (!response.ok) {
+            throw new Error(`Someting went wrong, Status code ${response.status}`)
+        }
+        return await response.json();
+    }
+
+
+
+    const updateUser = async (id, obj) => {
+        const update = await fetch(`${url}/${id}`, {
+            method: 'PATCH',
+            body: JSON.stringify(obj),
+              headers: {
+                'Content-type': 'application/json; charset=UTF-8',
+              },
+        });
+        
+        if (!update.ok) {
+            throw new Error(`Someting went wrong, Status code ${update.status}`)
+        }
+        return await update.json();
+    }
+
+
+
+
+    getAllUsers(url)
+        .then(value => value.reverse())
+        .then(data => {
+            paginationUser(data); 
```

Comment:
pagination should remain after filtration 
![image](https://user-images.githubusercontent.com/84564899/126711289-53c7b430-eb55-4b60-bb94-63b49eff1f72.png)